.github/workflows/edr-ci.yml

```diff
@@ -10,7 +10,6 @@ on:
   workflow_dispatch:
 
 env:
-  RUSTFLAGS: -Dwarnings
```

Comment:
This should not have any adverse side-effects, as we also manually pass this argument to `cargo clippy` which should fail if there are any warnings.

For reference:

`cargo clippy --workspace --all-targets --all-features -- -D warnings`

Reply:
Thanks for the explanatory comment! I'm wondering if removing it has effect on the `check-edr` and `test-edr-rs`? E.g. if there are some unused imports with some feature combination, would the CI not fail now? It's not a merge blocker for me if it doesn't, just want to make sure we consider it.

Reply:
Good catch! Given that we use `cargo hack` in `check-edr`, we would lose some functionality - I think. I've re-added the environment variables for that specific command [here](https://github.com/NomicFoundation/edr/pull/664/commits/65a545eb8047400d473b883db7c68d2d524bf043).

Clippy catches all warnings that `cargo check|build|test` would catch, as it is a superset of the check command - under the hood. Since we run `cargo clippy` with `--all-targets`, the test target is also run.

This means that it's possible for the `test-edr-rs` step to succeed with warnings, but the step running `cargo clippy` would fail the CI.

To validate this, I just added two simple functions (dead code) to `lib.rs`:

```rs
fn foo() {
    println!("foo");
}

#[cfg(test)]
fn bar() {
    prinltn!("bar");
}
```

Running `cargo check --all-targets` gives the following warnings:

```
warning: function `foo` is never used
  --> crates/edr_eth/src/lib.rs:62:4
   |
62 | fn foo() {
   |    ^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: `edr_eth` (lib) generated 1 warning
warning: function `bar` is never used
  --> crates/edr_eth/src/lib.rs:67:4
   |
67 | fn bar() {
   |    ^^^

warning: `edr_eth` (lib test) generated 2 warnings (1 duplicate)
```

These become errors when running `cargo clippy --all-targets -- -D warnings` (failing one at a time):

```
error: function `foo` is never used
  --> crates/edr_eth/src/lib.rs:62:4
   |
62 | fn foo() {
   |    ^^^
   |
   = note: `-D dead-code` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(dead_code)]`

error: could not compile `edr_eth` (lib) due to 1 previous error

error: function `bar` is never used
  --> crates/edr_eth/src/lib.rs:63:4
   |
63 | fn bar() {
   |    ^^^
   |
   = note: `-D dead-code` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(dead_code)]`
```